Exec/science/planet/problem_initialize_state_data.H

```diff
@@ -0,0 +1,174 @@
+#ifndef problem_initialize_state_data_H
+#define problem_initialize_state_data_H
+
+#include <prob_parameters.H>
+#include <eos.H>
+#include <model_parser.H>
+#include <ambient.H>
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void problem_initialize_state_data (int i, int j, int k,
+                                    Array4<Real> const& state,
+                                    const GeometryData& geomdata)
+{
+
+    const Real* dx = geomdata.CellSize();
+    const Real* problo = geomdata.ProbLo();
+
+    Real coord[3] = {0.0};
+
+    coord[0] = problo[0] + dx[0] * (static_cast<Real>(i) + 0.5_rt);
+#if AMREX_SPACEDIM >= 2
+    coord[1] = problo[1] + dx[1] * (static_cast<Real>(j) + 0.5_rt);
+#endif
+#if AMREX_SPACEDIM == 3
+    coord[2] = problo[2] + dx[2] * (static_cast<Real>(k) + 0.5_rt);
+#endif
+
+    Real shear_height_r = problem::shear_height * dx[2];
+
+    int int_shear_width_x = static_cast<int>(problem::shear_width_x);
+    int int_shear_width_y = static_cast<int>(problem::shear_width_y);
+    Real velocity_gradient = problem::shear_amplitude / static_cast<Real>(problem::shear_height);
+
+    int shear_bottom_index = static_cast<int>(problem::shear_height_loc / dx[2]) - problem::shear_height;
```

Comment:
This doesn't work in 2D, we should probably fix that at some point if we want to run in fewer dimensions.